library/core/src/fmt/mod.rs

```diff
@@ -633,6 +633,23 @@ pub use macros::Debug;
 /// [tostring]: ../../std/string/trait.ToString.html
 /// [tostring_function]: ../../std/string/trait.ToString.html#tymethod.to_string
 ///
+/// # Internationalization
+///
+/// Because a type only has one `Display` implementation, it is often preferable
+/// to only implement `Display` when there is a single most "obvious" way that
+/// values can be formatted as text. This could mean formatting according to the
+/// "invariant" culture and "undefined" locale, or it could mean that the type
+/// display is designed for a specific culture/locale, such as developer logs.
```

Comment:
Does a mention of `DIsplay`-related modifiers make sense here. A strict reading here makes me feel like `i32` wouldn't have a `Display` because there's no "obvious" "base 10 is better than any other base" rule. Instead, these alternate formats are satisfied with modifiers. I see the paragraph below mentions these other traits; maybe something like:
```suggestion
/// Because a type only has one `Display` implementation, it is often preferable
/// to only implement `Display` when there is a single most "obvious" way that
/// values can be formatted as text besides [the standard set of alternative formats][formatting traits]. This could mean formatting according to the
/// "invariant" culture and "undefined" locale, or it could mean that the type
/// display is designed for a specific culture/locale, such as developer logs.
```

Reply:
That base-10 is the "obvious" textual representation for numbers comes from the same ambient understanding that makes the binding/associativity of the infix math operators "obvious" – it's taught that way in grade school. Plus, the `Display` representation of numbers is that which is prescribed by the invariant culture, so falls under that concept of an "obvious" default format as well, even if Rust gets used in an environment where a different base is pervasively conventional.

Thus I believe this case is sufficiently covered by the immediately following mention of the alternative formatting traits plus quite thorough module-level documentation on the `format_args!` syntax, and doesn't need further call-out here.

It's intended to be somewhat vague here and avoid prescribing anything as fully "correct" or "incorrect," thus only showing examples of what could be considered "obvious," never setting a hard requirement. If anything, numbers not perfectly fitting could help reinforce that this is a conventional suggestion, not a requirement.

Reply:
Rephrasing: your wording suggests that a type whose canonical encoding fits one of the alternative formatting traits should only implement that one and not `Display`. Real usage suggests the opposite: that it would implement `Display` for the canonical encoding and not the other trait, as that encoding is not alternate.